### ISSUE TITLE
508-defect-2 [AXE-CORE]: Heading levels should increase by one - Facility health services heading level 

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#15701]
## Description
As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.

There are two separate heading issues that need to be addressed in this work. 

**From the style guide:** 

Headings and accessibility
- Always use semantic headings, logically nested in the proper order. Do not use improper nesting to achieve a different font size. In other words, an h1 should never be followed by an h3; it should only be followed by an h2. Use the font-size utility to change the font size of a semantic heading instead.

---

**1. Appointments should be the same H level as Contact information** - reported by Ryan from design/style perspective 

<details>
<summary> Design spec </summary>

![image](https://user-images.githubusercontent.com/55411834/102665616-5edb2800-4142-11eb-98be-c4a97262604d.png)

</details>

---

**2. Skipping levels when there is no H4 location name** - reported by Jennifer as [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

<details>
<summary> Location names, such as Center for Treatment of Addictive Disorders, (seen within Addiction and Substance Abuse Treatment), use H4 headings, making the H5 for **Hours** and Phone headings appropriate. </summary>

![image](https://user-images.githubusercontent.com/55411834/97904272-7890ef00-1cfd-11eb-9ead-3665e64916ee.png)
![image](https://user-images.githubusercontent.com/55411834/97904472-c4dc2f00-1cfd-11eb-85c1-af4626c7f319.png)
![image](https://user-images.githubusercontent.com/55411834/97904135-5008f500-1cfd-11eb-83f8-5993f41e8aa4.png)

</details>

<details>
<summary>Without this H4 location name heading,  **Hours** and Phone headings skip a level. </summary>

![image](https://user-images.githubusercontent.com/55411834/97905387-1b963880-1cff-11eb-9361-a1440b658cfa.png)

</details>

See [Common A11y Issues - Headings documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/common-issues/headings-should-increase-by-one.md)

## Point of Contact

**VFS Point of Contact:** Jennifer

## Screenshots or Trace Logs

<details>
<summary> Area where the heading level issue exists </summary>

![image](https://user-images.githubusercontent.com/55411834/97920920-6cb22680-1d17-11eb-9a12-2eac3ea81e69.png)

</details>

## Tasks
- [ ] Resolve heading level defects occurring within facility health service accordions
- [x] Heading level for Appointments is H3
- [x] When there is no Location name specified, heading levels do not skip a level
- [ ] Modify e2e tests to ensure that all of the accordions are opened and the axe checks are run on the initially hidden content
(Note: other axe errors, such as content landmarks and color contrast, are beyond the scope of this issue)

**Comment by davidconlon** [comment]
@mmiddaugh not sure how to resolve this one.
Not all health services will have a Center name. Some are just address.
All will have hours, etc.

**Comment by mmiddaugh** [comment]
Scheduled time to talk with Jennifer on 11/11

**Comment by jenstrickland** [comment]
You could make them all `h4` and apply a utility class for the ones for Hours and Phone.

**Comment by mmiddaugh** [comment]
This will be addressed in https://github.com/department-of-veterans-affairs/va.gov-team/issues/17755

**Comment by mpelzsherman** [comment]
Fixed in https://github.com/department-of-veterans-affairs/vets-website/pull/15705

**Comment by mmiddaugh** [comment]
@joshkimux this is ready for your validation when you are available

**Comment by joshkimux** [comment]
@mmiddaugh Awesome!  🥳 

Two steps I think remain here!
- [Accordion titles should be added as `h3`s following guidance in ticket 18913](https://github.com/department-of-veterans-affairs/va.gov-team/issues/18913). This will help screen reader users distinguish the various contact information headings from one another when navigating by headings.
- After adding the accordion titles, "Phone" beneath "Appointment" should remain a `h5` as "Appointment" will become an `h4`

I recommend using the headingsmap tool and imagining it as an outline or table of contents of the page. Repeated headings (as seen in the first screenshot), would be confusing since you might wonder, "Why are there so many Contact Information sections? Are these all related to each other?" Adding in the accordion titles would fix that as you'd realize that the contact information sections are in fact for different topics.

<img width="1555" alt="Screen Shot 2021-01-27 at 9 51 45 AM" src="https://user-images.githubusercontent.com/14154792/106009083-fa3ab380-6085-11eb-843b-fc76b50f384d.png">
<img width="1792" alt="Screen Shot 2021-01-27 at 10 00 33 AM" src="https://user-images.githubusercontent.com/14154792/106009765-b300f280-6086-11eb-9329-582245024286.png">

**Comment by mmiddaugh** [comment]
This issue is returning to the backlog based on a larger conversation about headings

**Comment by mmiddaugh** [comment]
Closing this as remaining work is in #18913
